test/token/WstEthL2Token.t.sol

```diff
@@ -0,0 +1,267 @@
+import {Upgrades} from "script/lib/Upgrades.sol";
+import {DefenderOptions} from "@openzeppelin/foundry-upgrades/Options.sol";
+import {WstEthL2Token} from "src/token/WstEthL2Token.sol";
+import {WstEthL2TokenHarness} from "test/token/WstEthL2TokenHarness.sol";
+import {WstEthL2TokenV2Fake} from "test/token/WstEthL2TokenV2Fake.sol";
+import {OwnableUpgradeable} from
+    "openzeppelin-contracts-upgradeable/contracts/access/OwnableUpgradeable.sol";
+import {INttToken} from "@wormhole-foundation/native_token_transfer/interfaces/INttToken.sol";
+
+contract WstEthL2TokenTest is Test {
+    WstEthL2TokenHarness token;
+    address minter = makeAddr("NTT");
+    address governance = makeAddr("Governance");
+    uint256 MAX_INT = 2 ** 256 - 1;
+
+    function setUp() public virtual {
+        address proxy = Upgrades.deployUUPSProxy(
+            "out/ERC1967Proxy.sol/ERC1967Proxy.json",
+            "WstEthL2TokenHarness.sol",
+            abi.encodeCall(WstEthL2Token.initialize, ("Wrapped Staked Eth", "wstEth", governance))
+        );
+        vm.label(proxy, "Proxy");
+
+        token = WstEthL2TokenHarness(proxy);
+        vm.prank(governance);
+        token.setMinter(minter);
+        vm.label(address(token), "WstEthL2Token");
+    }
+
+    // This internal method supplies options to `Upgrades.upgradeProxy`. For more information, see documentation in Options.sol
+    function _upgradeProxyOpts() public returns (Options memory) {
+        return Options({
+            unsafeSkipAllChecks: vm.envOr("SKIP_SAFETY_CHECK_IN_UPGRADE_TEST", false),
+            referenceContract: "",
+            constructorData: "",
+            unsafeAllow: "",
+            unsafeAllowRenames: false,
+            unsafeSkipStorageCheck: false,
+            defender: DefenderOptions(false, false, "", bytes32(""), "")
+        });
+    }
+
+    // Initialization and Upgrade tests
+
+    function testFuzz_CorrectlyInitializesTheToken(
+        string memory _name,
+        string memory _symbol
+    ) public {
+        address proxy = Upgrades.deployUUPSProxy(
+            "out/ERC1967Proxy.sol/ERC1967Proxy.json",
+            "WstEthL2TokenHarness.sol",
+            abi.encodeCall(WstEthL2Token.initialize, (_name, _symbol, governance))
+        );
+        vm.label(proxy, "Proxy");
+
+        token = WstEthL2TokenHarness(proxy);
+        vm.label(address(token), "WstEthL2Token");
+
+        assertEq(token.symbol(), _symbol);
+        assertEq(token.name(), _name);
+
+        // Current balance is 0
+        assertEq(token.totalSupply(), 0);
+
+        // Check Roles are set up correctly
+        assertEq(token.owner(), governance);
+        assertEq(token.minter(), address(0));
+    }
+
+    function testFuzz_InitializedTokenOwnerCanSetAndChangeMinter(
+        address _newMinter,
+        uint256 _amount
+    ) public {
+        _amount = bound(_amount, 1, MAX_INT);
+        vm.assume(_newMinter != address(0));
```

Comment:
This test fails if `_newMinter == governance`. Either we should update the assume or update the test logic. Technically governance could also be the minter so I think the latter is better

Reply:
Nice catch, this is an error in the test logic. Will update this test.